homeassistant/components/modbus/validators.py

```diff
@@ -241,7 +242,10 @@ def duplicate_entity_validator(config: dict) -> dict:
                     addr += "_" + str(entry[CONF_COMMAND_ON])
                 if CONF_COMMAND_OFF in entry:
                     addr += "_" + str(entry[CONF_COMMAND_OFF])
-                addr += "_" + str(entry.get(CONF_SLAVE, 0))
+                inx = entry.get(CONF_SLAVE, None)
+                if not inx:
+                    inx = entry.get(CONF_DEVICE_ADDRESS, 0)
```

Comment:
```suggestion
                inx = entry.get(CONF_SLAVE, None) or entry.get(CONF_DEVICE_ADDRESS, 0)
```

Reply:
Good catch, changed !! to be honest I actually thought your change would yield a boolean as result.